src/Microsoft.TestPlatform.CoreUtilities/Utilities/MulticastDelegateUtilities.cs

```diff
@@ -57,17 +57,22 @@ public static void SafeInvoke(this Delegate delegates, object sender, object arg
                 try
                 {
                     handler.DynamicInvoke(sender, args);
-                    EqtTrace.Verbose("MulticastDelegateUtilities.SafeInvoke: {0}: Invoking callback {1}/{2} for {3}.{4}, took {5} ms.",
-                            traceDisplayName,
-                            ++i,
-                            invocationList.Length,
-                            handler.GetTargetName(),
-                            handler.GetMethodName(),
-                            stopwatch.ElapsedMilliseconds);
+                    if (EqtTrace.IsVerboseEnabled)
```

Comment:
Chceck early if we should even log, before building that big string.

Reply:
Is it really impacting performance?

Reply:
Yes, even though by very little.